swarm/storage/localstore/mode_put.go

```diff
@@ -18,18 +18,32 @@ package localstore
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/ethereum/go-ethereum/metrics"
 	"github.com/ethereum/go-ethereum/swarm/chunk"
 	"github.com/ethereum/go-ethereum/swarm/shed"
+	"github.com/ethereum/go-ethereum/swarm/spancontext"
+	olog "github.com/opentracing/opentracing-go/log"
 	"github.com/syndtr/goleveldb/leveldb"
 )
 
 // Put stores the Chunk to database and depending
 // on the Putter mode, it updates required indexes.
 // Put is required to implement chunk.Store
 // interface.
-func (db *DB) Put(_ context.Context, mode chunk.ModePut, ch chunk.Chunk) (exists bool, err error) {
-	return db.put(mode, chunkToItem(ch))
+func (db *DB) Put(ctx context.Context, mode chunk.ModePut, ch chunk.Chunk) (exists bool, err error) {
+	ctx, sp := spancontext.StartSpan(ctx, fmt.Sprintf("localstore.Put.%s", mode))
```

Comment:
Considering the number of `Get` and `Put` and `Has` operations that we have (probably A LOT), we should def utilize counters and resetting timers, as they scale very well - they are a constant number. Counters are already here, but I see we are missing ResettingTimers.

Also since we have a lot of these, I am not sure we want to add tracing spans - each of this will appear on a trace of a specific request, which probably won't scale very well. We can try it and see what the result is, but I have the feeling that we will have to have a way to disable OpenTracing trace spans.

Reply:
Yes, this is why I have asked for a bit more specifics on required metrics and traces. I was following your advices on https://gitter.im/ethersphere/hq?at=5cb4879f3ebbdc55b3c02f26. 

Could you point what do you refer to as ResettingTimers?

Reply:
If you grep for `ResettingTimer` in the codebase you will see them used.

I added specifics in the chat room when you asked:

```
can you provide a list of what metrics and traces are required to be implemented? - all metrics and traces that would help us establish a baseline on the performance of the localstore when we run it on the nodes we host
get/put counters and resetting timers is the most basic ones
counters for each index as in number of operations would also help i guess
you should have a better idea then me, since you know the intrinsic impl.
```

`traces` can mean OpenTracing, or `log.Trace`.

Reply:
@nonsense I have added resetting timer metrics and organized the metrics code just a bit better.